examples/mlx_metal_kernel_opt/evaluator.py

```
@@ -0,0 +1,1395 @@
+            for i, config in enumerate(baseline_configs, 1):
+                print(f"  [{i}/{len(baseline_configs)}] GPU-protected baseline: {config.name}")
+
+                retry_count = 0
+                while retry_count <= self.max_retry_attempts:
+                    try:
+                        # Clean GPU state before each attempt
+                        self._ensure_clean_gpu_state()
+
+                        # Run with GPU protection
+                        success, result = self._bulletproof_execute_with_gpu_protection(
+                            lambda: self.benchmark_suite.run_single_benchmark(config)
```

Ruff (0.11.9)

Function definition does not bind loop variable `config`

(B023)